### PR TITLE
Reader: Pull suggestions from subscribed tags, Add more suggestions, Save Space

### PR DESCRIPTION
**Pull request by blowery**
Bump the suggestion provider out to a separate HoC and use it from search and following.

### CODE REVIEW

File: client/reader/search-stream/suggestion-provider.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { map, sampleSize } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import TagSubscriptions from 'lib/reader-tags/subscriptions';
+import i18nUtils from 'lib/i18n-utils';
+import { suggestions } from 'reader/search-stream/suggestions';
+
+export default ( Element, count = 3 ) => {
```
Comment (samouri) [inline client/reader/search-stream/suggestion-provider.jsx]: why did you move this back to a HoC?

File: client/reader/search-stream/suggestion-provider.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			TagSubscriptions.off( 'change', this.handleChange );
+		}
+
+		suggestionsFromTags() {
+			const tags = TagSubscriptions.get();
+			if ( tags ) {
+				if ( tags.length <= count ) {
+					return [];
+				}
+				return map( sampleSize( tags, count ), tag => ( tag.display_name || tag.slug ).replace( /-/g, ' ' ) );
```
Comment (samouri) [inline client/reader/search-stream/suggestion-provider.jsx]: do all tags have display_name? why are some missing them

**Comment by samouri** [comment]
I like this solution 👍

File: client/reader/search-stream/suggestion-provider.jsx (hunk 1 above)
Comment (blowery) [inline client/reader/search-stream/suggestion-provider.jsx]: because this is now waiting for the tags to load before making the decision.

File: client/reader/search-stream/suggestion-provider.jsx (hunk 2 above)
Comment (blowery) [inline client/reader/search-stream/suggestion-provider.jsx]: i think it may be hard-coded to the slug? I'm mostly matching how we show them in the sidebar.

**Comment by blowery** [comment]
This also now adds suggestion fallbacks for Spanish, German, and French.

**Comment by samouri** [comment]
I'll test out the latest changes! code LGTM

**Comment by samouri** [comment]
Looks great for supported langs:
screen shot 2017-03-01 at 2 38 21 pm

A bit awkward of a space for unsupported langs:
screen shot 2017-03-01 at 2 39 34 pm
